plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -0,0 +1,409 @@
+   * correct HadoopSecurityManager Java class
+   * 
+   * @param props
+   * @param log
+   * @return a HadoopSecurityManager object. Will throw exception if any errors occur (including not
+   *         finding a class)
+   * @throws RuntimeException
+   *           : If any errors happen along the way.
+   */
+  public static HadoopSecurityManager loadHadoopSecurityManager(Props props, Logger log)
+          throws RuntimeException {
+
+    Class<?> hadoopSecurityManagerClass = props.getClass(HADOOP_SECURITY_MANAGER_CLASS_PARAM, true,
+            HadoopJobUtils.class.getClassLoader());
+    log.info("Loading hadoop security manager " + hadoopSecurityManagerClass.getName());
+    HadoopSecurityManager hadoopSecurityManager = null;
+
+    try {
+      Method getInstanceMethod = hadoopSecurityManagerClass.getMethod("getInstance", Props.class);
+      hadoopSecurityManager = (HadoopSecurityManager) getInstanceMethod.invoke(
+              hadoopSecurityManagerClass, props);
+    } catch (InvocationTargetException e) {
+      log.error("Could not instantiate Hadoop Security Manager "
+              + hadoopSecurityManagerClass.getName() + e.getCause());
+      throw new RuntimeException(e.getCause());
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new RuntimeException(e.getCause());
+    }
+
+    return hadoopSecurityManager;
+
+  }
+
+  /**
+   * Fetching token with the Azkaban user
+   * 
+   * @param hadoopSecurityManager
+   * @param props
+   * @param log
+   * @return
+   * @throws HadoopSecurityManagerException
+   */
+  public static File getHadoopTokens(HadoopSecurityManager hadoopSecurityManager, Props props,
+          Logger log) throws HadoopSecurityManagerException {
+
+    File tokenFile = null;
+    try {
+      tokenFile = File.createTempFile("mr-azkaban", ".token");
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new HadoopSecurityManagerException("Failed to create the token file.", e);
+    }
+
+    hadoopSecurityManager.prefetchToken(tokenFile, props, log);
+
+    return tokenFile;
+  }
+
+  /**
+   * <pre>
+   * If there's a * specification in the "jar" argument (e.g. jar=./lib/*,./lib2/*),
+   * this method helps to resolve the * into actual jar names inside the folder, and in order.
+   * This is due to the requirement that Spark 14 doesn't seem to do the resolution for users
+   * 
+   * TODO: test case required: one replacement, two replacement.
+   * </pre>
+   * 
+   * @param unresolvedJarSpec
+   * @return jar file list, comma separated, all .../* expanded into actual jar names in order
+   * 
+   */
+  public static String resolveWildCardForJarSpec(String workingDirectory, String unresolvedJarSpec,
+          Logger log) {
+
+    log.info("resolveWildCardForJarSpec: unresolved jar specification: " + unresolvedJarSpec);
+    log.info("working directory: " + workingDirectory);
+
+    if (unresolvedJarSpec == null || unresolvedJarSpec.isEmpty())
+      return null;
```

Comment:
Does it make more sense to return empty string rather than null?  If so, please document that the returned value can be null